File: src/main/kotlin/com/zdan91/duiker/utils/SecurityUtils.java
package com.zdan91.duiker.utils;

/**
 * @author hookszhang
 */
public class SecurityUtils {

    private static ThreadLocal<Integer> threadLocal = new ThreadLocal<>();

    /**
     * 获得当前登录人 userId
     *
     * @return 如果未登录，返回 -1
     */
    public static int getCurrentUserId() {
        return threadLocal.get();
    }

    public static void setCurrentUserId(int userId) {
        threadLocal.set(userId);
    }

    public static void unset() {
        threadLocal.remove();
    }
}

File: src/main/java/com/zdan91/duiker/utils/SecurityUtils.kt
package com.zdan91.duiker.utils

import com.zdan91.duiker.bean.User

/**
 * Spring Security JWT Utils.
 *
 * @author hookszhang
 */
object SecurityUtils {

    private val threadLocal = ThreadLocal<User>()

    /**
     * 从 ThreadLocal 中获得当前登录用户
     */
    fun currentLoginUser() : User? {
        return threadLocal.get()
    }

    /**
     * 将当前登录用户放入 ThreadLocal 中
     */
    fun putCurrentLoginUser(user: User) {
        threadLocal.set(user)
    }

    /**
     * Check if a user is authenticated.
     */
    fun isAuthenticated() : Boolean {
        return threadLocal.get() != null
    }
}
